docs/data/upload.md

```diff
@@ -0,0 +1,113 @@
+- Run a Golang script calling the data management service `sync` method.
+- Upload images from your mobile device using the Viam mobile app.
+
+## Upload data with a Python script
+
+You can use the Viam Python SDK's data client API [`file_upload_from_path`](/build/program/apis/data-client/#fileuploadfrompath) method to upload each file you'd like to upload from your computer to the Viam cloud.
+
+1. Install the [Viam Python SDK](https://python.viam.dev/) by running the following command on the computer from which you want to upload data:
+
+   ```sh {class="command-line" data-prompt="$"}
+   pip install viam-sdk
+   ```
+
+2. Create a Python script file in a directory of your choice and [add code to establish a connection](/build/program/apis/data-client/#establish-a-connection) from your computer to your [Viam app](https://app.viam.com) {{< glossary_tooltip term_id="location" text="location" >}} or individual {{< glossary_tooltip term_id="part" text="machine part" >}}.
+
+3. To upload just one file, make a call to the `file_upload_from_path` method according to [the data client API documentation](/build/program/apis/data-client/#fileuploadfrompath).
+   The following example code could be placed inside the `main()` function (or a function called from `main()`):
+
+   ```python {class="line-numbers linkable-line-numbers"}
+   await data_client.file_upload_from_path(
+     # The ID of the machine part the file should be associated with
+     part_id="abcdefg-1234-abcd-5678-987654321xyzabc",
+     # Any tags you want to apply to this file
+     tags=["cat", "animals", "brown"],
+     # Path to the file
+     filepath="/Users/Artoo/my_cat_photos/brown-cat-on-a-couch.png"
+   )
+   ```
+
+   To upload all the files in a directory, you can use the same [`file_upload_from_path`](/build/program/apis/data-client/#fileuploadfrompath) method inside a `for` loop, for example:
+
+   ```python {class="line-numbers linkable-line-numbers"}
+   import os # Add this package up at the top with your other imports
+
+   my_data_directory = "/Users/Artoo/my_cat_photos"
+
+   for file_name in os.listdir(my_data_directory):
+     await data_client.file_upload_from_path(
+       part_id="abcdefg-1234-abcd-5678-987654321xyzabc",
+       tags=["cat", "animals", "brown"],
+       filepath=os.path.join(my_data_directory, file_name)
+     )
+   ```
+
+4. Save and run your code once.
+   Running your code more than once will duplicate the data.
+   View your uploaded data in your [**DATA** page in the Viam app](https://app.viam.com/data/view).
+
+## Sync data with the `sync` method
+
+The Viam Golang SDK includes a data service API with a [`sync`](/data/#sync) method that syncs all the data from your machine to the cloud.
+The [cloud sync tools](/data/cloud-sync/) sync data automatically according to how you configure cloud sync, but if you prefer to sync programatically, you can instead run a script that calls the `sync` method.
```

Comment:
Supernit OPT: tiny clarity addition.

```suggestion
The [cloud sync tools](/data/cloud-sync/) sync data automatically according to how you configure cloud sync, but if you prefer to sync programatically, you can instead run a script that calls the `sync` method directly.
```

Reply:
Hmm. I wonder if this could be interpreted in less-clarifying ways, like it makes me wonder "oh, does the data service use the `sync` method indirectly? What does this mean?"

Reply:
Ah, gotcha. Fair point!